/1. What will be output of printing following calculations
a. 1.5e200 *2.0e210
b. 3.4e-320 / 1e100
xplain the output/7+++++++++++7748512*/79846512100000000000010514-
4/-610-*6+01*154-40.1-40..1-1414141414141414141414141414141414141414141414141414141414141414141414141414141414141414141414.06314*-0...22

In [3]:
1.5**200 *2.0**210

2.7198872460889578e+98

In [5]:
3.4**-320/1**100

8.447856857801338e-171

Q2. What is output of printing following
a. 1/3
b. format(1/3, '.3f')
c. format(1/3, '.1f')
d. 6*1/3
e. 1/3 + 1/3 + 1/3 + 1/3 + 1/3 + 1/3


In [6]:
1/3

0.3333333333333333

In [9]:
format(1/3, '.3f')

'0.333'

In [10]:
format(1/3, '.1f')

'0.3'

In [11]:
6*1/3

2.0

In [12]:
1/3 + 1/3 + 1/3 + 1/3 + 1/3 + 1/3

1.9999999999999998

Q3. In above question what is difference between output of a and b ? also, d and e?


In the above question in a, the digit next to the decimal point are more and in b there is limit for decimal that is 3 
In d first  there is division then multiplication and in e is first there is division then addition


Q4. Study how format function works to print floating point values and strings . Then solve following using format function
a. print only first 5 decimal values of 22/7 (hint: see b of Q3)
b. print Hello string with 5 spaces on right side, 'Hello     ' ( hint: use <)
c. print Hello string with 5 spaces on left side, '     Hello' ( hint: use <) 
d. print Hello string with 5 '#' on left side and 5 '#' on right side, '     Hello     ' ( hint: use ^ and # as fill value)
e. print 10 spaces (hint: give 10 as argument)

In [14]:
format(22/7, '.3f')

'3.143'

In [15]:
print(format('HELLO', '<10'))


HELLO     


In [16]:
print(format('HELLO', '>10'))


     HELLO


In [17]:
print(format('hello', '^15'))


     hello     


In [18]:
print(format(' ', '^10'))


Q5. We can use ord() to print ASCII value of single character and chr() to convert given ASCII value to character. 
a. find ASCII of characters'a' , 'A' , 'z' , 'Z' using ord() one by one
b. find ASCII of characters '0', '9', ' ', "\n" , "\t" using ord() one by one
c. Find characters for ASCII values 35, 67, 50, 99
d. Where ASCII value begin for lower case characters where it ends? also find same for digits and upper case characters

In [19]:
print(ord('a'))


97


In [20]:
print(ord('A'))


65


In [21]:
print(ord('Z'))


90


In [22]:
print(ord('z'))


122


In [25]:
print(ord('0'))
print(ord('9'))




48
57


In [26]:
print(ord(' '))

32


In [27]:
print(ord('\n'))

10


In [28]:
print(ord('\t'))

9


In [29]:
print(chr(35))
print(chr(67))
print(chr(50))
print(chr(99))


#
C
2
c


For lower case value begins from 97 to 122
For upper case value begins from 65 to 90


Q6. Find how special chracters work inside print function
a. print("abc","\n","bcd")
b. print("abc\nbcd")
c. print(1,'\n',2)
d. print("hello \n\n!!")
e. print("This is '\t' means tab")
f. How many spaces are printed using \t?
g. print("This","is","different",sep='\t')
h. print("This","is","different",sep='#')
i. print("This","is","different",sep='\n') 
j. print("This","is","different",end='\n')


In [30]:
print("abc","\n","bcd")

abc 
 bcd


In [31]:
print("abc\nbcd")

abc
bcd


In [32]:
print(1,'\n',2)

1 
 2


In [33]:
print("hello \n\n!!")

hello 

!!


In [34]:
print("This is '\t' means tab")


This is '	' means tab


4

In [35]:
print("This","is","different",sep='\t')

This	is	different


In [36]:
print("This","is","different",sep='#')

This#is#different


In [37]:
print("This","is","different",sep='\n')

This
is
different


In [38]:
print("This","is","different",end='\n')

This is different


Q7. What is output of following statements and understand how operators work
a. 10 / 2
b. 10 // 2.0
c. 101 / 3
d. 101 // 3
e. 101 % 3
f. 2 ** 4
g. 3 ** 2
h. abs(-10) # give absolute value
i. divmod(101,3) # divmod gives two values , interger division and remainder
j. 8 << 1
k. 8 << 2
l. 32 >> 2
m. 16 >> 1


In [39]:
10/2

5.0

In [40]:
10//2

5

In [41]:
101/3

33.666666666666664

In [42]:
101//3

33

In [43]:
101%3

2

In [44]:
2**4

16

In [45]:
3**2

9

In [46]:
abs(-10)

10

In [47]:
divmod(101,3)

(33, 2)

In [48]:
8<<1

16

In [49]:
8<<2

32

In [50]:
32>>2

8

In [51]:
16>>1

8

Q8. Write a program to take marks of 3 subjects from a student. Calculate total in variable name 'total'.
(NOTE: Don't use sum as variable name, it is built in function in python)

In [52]:
n1= int(input("enter marks of first subject"))
n2= int(input("enter marks of second subject"))
n3= int(input("enter marks of third subject"))
total = n1+n2+n3
print(total)


enter marks of first subject50
enter marks of second subject70
enter marks of third subject80
200



Q9. Manually calculate output of following expressions using precedence and associativity rules 
and then verify output using python shell
a. 2 + 3 * (4 -1)
b. 2 + ((3 *4) - 8)
c. (2 + 3) * 4
d. -10 + 25 / (16 + 12)
e. 2 ** 2 ** 2
f. (3 ** 2) ** 3
g. x = y = 3 + 5
h. 8/2*(2+2)


In [53]:
2 + 3 * (4 -1)


11

In [54]:
2 + ((3 *4) - 8)

6

In [55]:
(2 + 3) * 4


20

In [56]:
-10 + 25 / (16 + 12)


-9.107142857142858

In [57]:
2 ** 2 ** 2

16

In [58]:
(3 ** 2) ** 3

729

In [60]:
x = y = 3 + 5


In [61]:
8/2*(2+2)

16.0

Q10. Check output of following 
a. x = y = 10
b. x1 = 20 = y1

In [62]:
x = y = 10

In [63]:
x1 = 20 = y1

SyntaxError: cannot assign to literal (1868644801.py, line 1)

Q11. What is output of following
a. x=10
y=34.4
print(x and y)
print(x or y)


In [64]:
x=10
y=34.4
print(x and y)
print(x or y)

34.4
10


### if else

1.	A student will not be allowed to sit in exam if his/her attendence is less than 75%.  
Take following input from user Number of classes held Number of classes attended. And print percentage of class attended  
Is student is allowed to sit in exam or not.  

In [65]:
class_held= int(input("enter the number of class held"))
class_attended =int(input("enter the number of class attended"))
total = (class_attended/class_held)*100
print(total)
if total < 75:
    print("student is not allowed to class")
else:
    print("student is allowed to class")


enter the number of class held80
enter the number of class attended66
82.5
student is allowed to class


2. accept amount from user and find the minimum number notes required to get the amount amount =512  
Notes: 2000,500,100,50,10,5,2,1  




In [71]:
amount=34267
notes_2k = amount // 2000
if notes_2k > 0:
    print("2000 -", notes_2k)
rem_amount= amount % 2000

notes_500 = rem_amount //500
if notes_500 > 0:
    print("500 -", notes_500)
rem_amount= amount % 500

notes_200 = rem_amount //200
if notes_200 > 0:
    print("200 -", notes_200)
rem_amount= amount % 200

notes_100 = rem_amount //100
if notes_100 > 0:
    print("100 -", notes_100)
rem_amount= amount % 100

notes_50 = rem_amount //50
if notes_50 > 0:
    print("50 -", notes_50)
rem_amount= amount % 50

notes_20, rem_amount = divmod(rem_amount,20)
if notes_20 > 0:
    print("20 -", notes_20)

    
notes_10, rem_amount = divmod(rem_amount,10)
if notes_10 > 0:
    print("10 -", notes_10)
notes_5, rem_amount = divmod(rem_amount,5)
if notes_5 > 0:
    print("5 -", notes_5)
notes_2, rem_amount = divmod(rem_amount,2)
if notes_2 > 0:
    print("2 -", notes_2)
notes_1, rem_amount = divmod(rem_amount,1)
if notes_1 > 0:
    print("1 -", notes_1)

2000 - 17
200 - 1
50 - 1
10 - 1
5 - 1
2 - 1


3.	Modify the above question Q1 to allow student to sit if he/she has medical cause. Ask user if he/she has medical cause or not ( 'Y' or 'N' ) and print accordingly.  
4.	A school has following rules for grading system:  
a.	Below 25 - F  
b.	25 to 45 - E  
c.	45 to 50 - D  
d.	50 to 60 - C  
e.	60 to 80 - B  
f.	Above 80 - A  
Ask user to enter marks and print the corresponding grade. 
marks =int(input("enter the marks"))


In [67]:
marks =int(input("enter the marks"))

if marks > 80:
    print("A")
elif 60 <= marks <= 80:
    print("B")
elif 50 <= marks <= 60:
    print("C")
elif 45 <= marks <= 50:
    print("D")
elif 25 <= marks <= 45:
    print("E")
else:
    print("F")


enter the marks78
B


5.	  If  x = 2 y = 5 z = 0  then find values of the following expressions and Print the output of following statements  
a.	x == 2  
b.	x != 5  
c.	x != 5 & y >= 5  
d.	z != 0 | x == 2  
e.	not(y < 10)

In [68]:
x=2
y=5
z=0
print(x==2)
print(x!=5)
print(x!=5 & y>=5)
print(z != 0 | x == 2 )
print(not(y < 10))


True
True
True
True
False


6.	Accept number from user and check whether it is divisible by 5 and 11 if divisible then display appropriate message.

In [70]:
num = int(input("enter the number"))
if num%11==0 & num%5==0:
    print("number is divisible by 11 and 5")
elif num%11==0:
    print("number is divisible by 11")
elif num%5==0:
    print("number is divisible by 5")
else:
    print("number is not divisible by 11 and 5")


enter the number55
number is divisible by 11 and 5


Write a program to calculate the electricity bill (accept number of unit from user) according to the following criteria :  
             Unit                                    Price    
First 100 units                                       no charge  
Next 100 units                                        Rs 5 per unit  
After 200 units                                       Rs 10 per unit  
(For example if input unit is 350 than total bill amount is Rs2000)  